contracts/staking/BaseShareField.sol

```diff
@@ -108,15 +107,12 @@ contract BaseShareField is DSMath {
 					accAmountPerShare) /
 					1e27 -
 					userInfo.rewardDebt; // Turn userInfo.amount to 18 decimals by multiplying tokenDecimalDifference if it's not and multiply with accAmountPerShare which is 27 decimals then divide it 1e27 bring it down to 18 decimals
-				uint256 rewardPerBlock = rdiv(pending, blocksToPay * 1e18); // bring both variable to 18 decimals so they would be in same decimals
+				uint256 rewardPerBlock = (pending * 1e27) / (blocksToPay * 1e18); // bring both variable to 18 decimals so they would be in same decimals
 				pending =
-					rmul(
-						(((firstMonthBlocksToPay * 1e27 * 5) / 10) + // multiply first month by 0.5x (5/10) since rewards in first month with multiplier 0.5 and multiply it with 1e27 to get it 27decimals
-							fullBlocksToPay *
-							1e27), // Multiply fullBlocksToPay with 1e27 to bring it to 27decimals
-						rewardPerBlock // rewardPerBlock is in 27decimals
-					) /
-					1e9; // Pending in 18 decimals so we divide 1e9 to bring it down to 18 decimals
+					((((firstMonthBlocksToPay * 1e27 * 5) / 10) + // multiply first month by 0.5x (5/10) since rewards in first month with multiplier 0.5 and multiply it with 1e27 to get it 27decimals
+						fullBlocksToPay *
+						1e27) * rewardPerBlock) / // Multiply fullBlocksToPay with 1e27 to bring it to 27decimals // rewardPerBlock is in 27decimals
+					1e36; // Pending in 18 decimals so we divide 1e36 to bring it down to 18 decimals
```

Comment:
I think we are doing extra 1e27 muls because at the end we divide by 1e16
so to me it seems we just need to multiply fullBlocks to pay by 1e18 and thats it, remove the two 1e27 and div by 1e36

Reply:
Since rewardPerBlock is 27 decimals we can't do it I think but there is no point to keep precision that high as well so I set multiply values to lower value so division value became smaller eventually too